ts/Series/TemperatureMap/TemperatureMapSeries.ts

```diff
@@ -0,0 +1,447 @@
+                }
+            },
+            /**
+             * Set of colors for each point. Can be an array of strings (if 3
+             * colors defined, every next marker is 33.3% smaller) or an array
+             * of color stops (e.g. for `[[0.2, '#ff0000'], [1, '#0000ff']]`
+             * the smallest top marker is 5 times smaller than the biggest
+             * bottom one).
+             *
+             * @sample maps/plotoptions/temperaturemap-themes
+             *         Temperature colors themes
+             * @sample maps/plotoptions/temperaturemap-color-steps
+             *         Temperature colors with steps
+             *
+             * @default ["#ff0000", "#ffff00", "#00ff00", "#00ffff", "#0000ff"]
+             * @type {Array<ColorType>}
+             */
+            temperatureColors: [
+                '#ff0000',
+                '#ffff00',
+                '#00ff00',
+                '#00ffff',
+                '#0000ff'
+            ]
+        }
+    );
+
+    /**
+     *
+     *  Functions
+     *
+     */
+
+    /**
+     * @private
+     */
+    public isPointInside(): true {
+        // Do not hide cut bubbles outside the plot area
+        return true;
+    }
+
+    /**
+     * @private
+     */
+    public getPxExtremes(): {
+        minPxSize: number,
+        maxPxSize: number
+    } {
+        const ret = super.getPxExtremes.apply(this),
+            temperatureColors = this.options.temperatureColors,
+            sizeFactor = this.sizeFactor || (1 / temperatureColors.length);
+
+        // Recalculate minPxSize to draw smaller bubbles
+        if (this.adjustMinSize) { // only when necessary
+            ret.minPxSize = ret.minPxSize * sizeFactor;
+        }
+
+        return ret;
+    }
+
+    /**
+     * Calculate radius of bubbles based on index of the color in the
+     * `series.temperatureColors` array.
+     * @private
+     */
+    public getRadius(
+        zMin: number,
+        zMax: number,
+        minSize: number,
+        maxSize: number,
+        value: number | null | undefined,
+        yValue?: number | null
+    ): number | null {
+        const temperatureColors = this.options.temperatureColors,
+            colorIndex = this.temperatureColorIndex || 0;
+
+        let ret = super.getRadius.apply(
+            this,
+            [zMin, zMax, minSize, maxSize, value, yValue]
+        );
+
+        if (!isNumber(ret)) {
+            return ret;
+        }
+
+        ret *= this.sizeFactor ?
+            this.sizeFactor :
+            1 - colorIndex / temperatureColors.length;
+
+        // Ceiling like in the bubble series - solves decimal issues and rounds
+        // to 1px
+        return Math.ceil(correctFloat(ret) * 2) / 2;
+    }
+
+    /**
+     * Perform animation on the bubbles
+     * @private
+     */
+    public animate(init?: boolean): void {
+        if (
+            !init &&
+            this.points.length < (this.options.animationLimit as any) // #8099
```

Comment:
Will `pick(this.options.animationLimit, Infinity)` do the same without `any` casting?